lib/auto-update.js

```diff
@@ -0,0 +1,81 @@
+// Adapted from famous-cli https://github.com/Famous/famous-cli/blob/master/lib/autoupdate.js
+'use strict';
+
+// Utility for colorizing terminal output
+var chalk = require( 'chalk' );
+
+// Cross-platform utility to invoke command-line processes
+var spawn = require( 'win-spawn' );
+
+// Tools for retrieving and validating NPM package release versions
+var semver = require( 'semver' );
+var latestVersion = require( 'latest-version' );
+
+// Details about the installed p5-cli package
+var current = require( '../package' ).version;
+
+/**
+ * update
+ *
+ * Run the update command
+ * @private
+ * @param {function} callback function to execute with the return code from the update command
+ * @returns {Promise} A promise that will complete when the update does
+ */
+function update( callback ) {
+  return new Promise(function( resolve, reject ) {
+    var child = spawn( 'npm', [ 'install', '-g', 'p5-cli' ]);
+
+    child.on( 'close', function( statusCode ) {
+      if ( statusCode === 0 ) {
+        console.log( 'Update successful!' );
+        resolve();
+      } else {
+        console.log([
+          '\n', chalk.red( 'Update failed.' ),
+          'It is possible you may need to run "npm install -g p5-cli" with\n',
+          '"sudo" (OSX) or "run as administrator" (Windows).\n',
+          'Continuing with the currently-installed version of p5...'
+        ].join( ' ' ));
```

Comment:
This error is logged because of an assumption that the most common installation error is a permissions issue. I will need to test what happens if *e.g.* the network cuts out half-way through an install; if it hoses the package, this may not be a good idea. Feedback desired!